CHANGELOG.md

```diff
@@ -8,6 +8,10 @@ Changelog](https://keepachangelog.com/en/1.1.0/), and this project adheres to
 
 ## [Unreleased]
 
+### Fixed
+
+- Introspection queries are now skipped when running `sqlmigrate`.
```

Comment:
> If it's not mentioned somewhere already, it might be worth saying that [...]

Thanks, I've added further clarification in this fixup 17052aa9c2cb179fdcb16aec960ecb8ae015f7b0

> whether we wrap the introspection and subsequent dependent DDL in a transaction manually to avoid potential (even if unlikely) racing?

I think we should avoid this, because we usually start the migration by
executing a DDL that has a very constricted lock, usually an AccessExclusive,
but that runs very fast. Then we move to a slow-running DDL that requires a
less constricted lock. Wrapping them in a transaction means retaining the
AccessExclusive lock until the end of the transaction.

But you have a point, this repo hasn't considered race conditions yet and we
should prevent it, even if that doesn't occur in Kraken due to the
single-pod-only constriction for the migrate command.

Thinking about this further, I think that a session-level advisory lock in the
`migrate_with_timeouts` command might be a good idea. It would prevent the
`migrate` command of running in parallel entirely